Psg 4895 use repo templates

jenn-macfarlane opened the pull request:
# Pull Request Template

## What's New?

Uses repo templates to format README, LICENSE, and remove community files in favour of using organization community files.


## Screenshots (if appropriate):

(The Passage logos will pick up theme change, this is just a screen shot from vscode):

Main README:
![CleanShot 2024-10-22 at 14 39 39](https://github.com/user-attachments/assets/4cdfa22f-0031-4bf4-851a-1c65c84b0d6f)

Complete:
![CleanShot 2024-10-22 at 09 47 24@2x](https://github.com/user-attachments/assets/c87597a8-c5dc-42af-b68f-04e63275590d)

Flex:
![CleanShot 2024-10-22 at 09 46 21@2x](https://github.com/user-attachments/assets/8c7dc858-1876-4186-b961-4a1a10ed7e98)

Authentikit:
![CleanShot 2024-10-22 at 09 47 39@2x](https://github.com/user-attachments/assets/d2f87322-c0d9-44d8-8d49-bc44e770b46f)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have manually tested my code thoroughly
- [x] I have added/updated inline documentation for public facing interfaces if relevant
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing integration and unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules


## Jira Ticket

Please add the Jira ticket number and link.

- [ ] Jira Ticket: [PSG-XXXX](https://passage-identity.atlassian.net/browse/PSG-XXXX)


## Additional context

Add any other context or screenshots about the pull request here.

jenn-macfarlane commented:
As of right now this PR is just updating the top-level README as the [passage](https://github.com/passageidentity/passage-android/blob/main/passage/generated/README.md)/[passageflex](https://github.com/passageidentity/passage-android/blob/main/passageflex/generated/README.md) readmes are auto-generated.

Couple questions:
- do we want to keep the auto-generated READMEs given they contain info the docs don't currently have (eg. "Documentation for API Endpoints" sections)?
- should I link to these auto-generated READMEs anywhere in the top-level one? My hunch is no as they're so specific

I'll also link to the appropriate anchor tag in the product tables in flutter/react-native where we reference Authentikit for ios/android and flex APIs.

agebhardt commented on README.md:
```diff
@@ -1,58 +1,127 @@
+ <sub><a href="https://docs.passage.id/home#passage-console">Learn more about Passage Console →</a></sub>
+</p>
 
 ### Passkey Complete
```
Like the way you broke these up! It looks like `Passkey Complete` is duplicated, though, and it appears twice in the README (in addition to a section for Flex and Authentikit)

jenn-macfarlane replied:
Good catch, thanks!

agebhardt reviewed (changes requested):
Looks like there's a duplicate of `Passkey Complete` sections (most visible when browsing the README in the branch:
Here: https://github.com/passageidentity/passage-android/tree/PSG-4895-use-repo-templates?tab=readme-ov-file#passkey-complete
And here: https://github.com/passageidentity/passage-android/tree/PSG-4895-use-repo-templates?tab=readme-ov-file#passkey-complete-1

They are also slightly different 😅 though I think the second initialize block better aligns with our pattern :)

jenn-macfarlane commented:
Hi @agebhardt! Thanks for the catch. I've cleaned that up, given the main README a bit of a change - namely including a "prerequisites" section under each product because they differ (either setting up Flex/Complete/either in Console). I've also added new READMEs with the same content to each individual package, and updated the product tables within each to point to the repo folder where they are found.

Please take a look and let me know if I missed anything/if it makes sense this way - I've got mobile's blessing to add these.

jenn-macfarlane commented on README.md:
```diff
@@ -1,58 +1,122 @@
+
+### In passage-android
+
+Use passage-android to implement [passage-android/passage](https://github.com/passageidentity/passage-android/passage) in your Android application to authenticate requests and manage users, [passage-android/passageflex](https://github.com/passageidentity/passage-android/passageflex) in your existing auth to use passkeys for registration, authentication, or added security on sensitive user actions ...
```
This is a mouthful but what I came up with that didn't sound super redundant given the above. Ppen to suggestions!

jenn-macfarlane commented on authentikit/README.md:
```diff
@@ -0,0 +1,70 @@
+![passage-android](https://storage.googleapis.com/passage-docs/github-md-assets/passage-android.png)
```
Have asked design for a new banner here.

jenn-macfarlane commented on authentikit/README.md:
```diff
@@ -0,0 +1,70 @@
+
+### In passage-android/authentikit
+
+Use passage-android/authentikit to implement Passkey Ready in your Android application to gather data on your users' passkey readiness.
```
Feel free to suggest an edit (should match what's in main readme).

jenn-macfarlane commented on passage/README.md:
```diff
@@ -0,0 +1,70 @@
+![passage-android](https://storage.googleapis.com/passage-docs/github-md-assets/passage-android.png)
```
Have asked design for a new banner here.

jenn-macfarlane commented on passageflex/README.md:
```diff
@@ -0,0 +1,70 @@
+![passage-android](https://storage.googleapis.com/passage-docs/github-md-assets/passage-android.png)
```
Have asked design for a new banner here.
